smarts/core/smarts.py

```diff
@@ -362,7 +362,7 @@ def done_agent_ids(dones):
         self._agent_manager.teardown_social_agents(agents_to_teardown)
         self._teardown_vehicles(vehicles_to_teardown)
 
-    def reset(self, scenario: Scenario) -> Dict[str, Observation]:
+    def reset(self, scenario: Scenario, start_time=0) -> Dict[str, Observation]:
```

Comment:
Perhaps it's worth adding a type annotation for the public interface? I see there are a few other places in this diff that are missing them as well, but I'll leave that up to you.
```suggestion
    def reset(self, scenario: Scenario, start_time: float = 0.0) -> Dict[str, Observation]:
```